### PR TITLE
Tooltip FIX The widget can be shown when showEvent is triggered not on the target (T818491)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: js/ui/popover.js [hunk 1]
```diff
@@ -91,13 +92,7 @@ var POPOVER_CLASS = "dx-popover",
             action({ event: e, target: $(e.currentTarget) });
         };
 
-        if(target.jquery || target.nodeType || typeUtils.isWindow(target)) {
-            that["_" + name + "EventHandler"] = undefined;
-            eventsEngine.on(target, eventName, handler);
-        } else {
-            that["_" + name + "EventHandler"] = handler;
-            eventsEngine.on(domAdapter.getDocument(), eventName, target, handler);
```
Comment (DokaRus) [inline js/ui/popover.js:99]: You missed the case when the target element is set as a selector. If this item is not in the DOM at the time of code execution, the Tooltip will not be shown

**Review by DokaRus — CHANGES_REQUESTED** [review]
See a note

File: js/ui/popover.js (hunk 1 above)
Comment (ksercs) [reply js/ui/popover.js:99]: Thanks, it's fixed now

Comment (ksercs) [reply js/ui/popover.js]: Thanks, it's fixed now

File: js/ui/popover.js [hunk 2]
```diff
@@ -91,12 +93,14 @@ var POPOVER_CLASS = "dx-popover",
             action({ event: e, target: $(e.currentTarget) });
         };
 
-        if(target.jquery || target.nodeType || typeUtils.isWindow(target)) {
-            that["_" + name + "EventHandler"] = undefined;
-            eventsEngine.on(target, eventName, handler);
-        } else {
-            that["_" + name + "EventHandler"] = handler;
+        var EVENT_HANDLER_NAME = `_${name}EventHandler`;
```
Comment (DokaRus) [inline js/ui/popover.js]: Template literals are part of ES6.

File: js/ui/popover.js [hunk 3]
```diff
@@ -108,10 +112,11 @@ var POPOVER_CLASS = "dx-popover",
 
         eventName = eventUtils.addNamespace(eventName, that.NAME);
 
-        if(that["_" + name + "EventHandler"]) {
-            eventsEngine.off(domAdapter.getDocument(), eventName, target, that["_" + name + "EventHandler"]);
+        var EVENT_HANDLER_NAME = `_${name}EventHandler`;
```
Comment (DokaRus) [inline js/ui/popover.js]: The same here

File: testing/tests/DevExpress.ui.widgets/tooltip.tests.js [hunk 4]
```diff
@@ -37,6 +41,46 @@ QUnit.test("render as tooltip", function(assert) {
     assert.ok(wrapper().length);
 });
 
+QUnit.test("tooltip should render when target is core renderer object", function(assert) {
```
Comment (DokaRus) [inline testing/tests/DevExpress.ui.widgets/tooltip.tests.js:44]: For new tests use ES6, pls.

// Remember to replace "require" with "import" at the beginning of the file

**Review by DokaRus — DISMISSED** [review]

**Review by DokaRus — APPROVED** [review]

**Review by mrjohn0011 — APPROVED** [review]

**Pull request merged**